CSharp/ArticleProjects/PersistOutputs/FileConventionsExample.cs

```diff
@@ -0,0 +1,107 @@
+            string jobId)
+        {
+            const string appPackageId = "PersistOutputsTask";
+            const string appPackageVersion = "1.0";
+
+            // Create and configure an unbound pool.
+            CloudPool pool = batchClient.PoolOperations.CreatePool(poolId: poolId,
+                virtualMachineSize: "small",
+                targetDedicatedComputeNodes: nodeCount,
+                cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));
+
+            // Specify the application and version to deploy to the compute nodes. You must
+            // first build PersistOutputsTask, then upload it as an application package.
+            // See https://azure.microsoft.com/documentation/articles/batch-application-packages/
+            pool.ApplicationPackageReferences = new List<ApplicationPackageReference>
+            {
+                new ApplicationPackageReference
+                {
+                    ApplicationId = appPackageId,
+                    Version = appPackageVersion
+                }
+            };
+
+            // Commit the pool to the Batch service
+            GettingStartedCommon.CreatePoolIfNotExistAsync(batchClient, pool).Wait();
+
+            CloudJob job = batchClient.JobOperations.CreateJob(jobId, new PoolInformation { PoolId = poolId });
+
+            // Create the blob storage container for the outputs.
+            job.PrepareOutputStorageAsync(linkedStorageAccount).Wait();
+
+            // Create an environment variable on the compute nodes that the
+            // task application can reference when persisting its outputs.
+            string containerName = job.OutputStorageContainerName();
+            CloudBlobContainer container = linkedStorageAccount.CreateCloudBlobClient().GetContainerReference(containerName);
+            string containerSas = container.GetSharedAccessSignature(CreateFullAccessPolicy());
+            string containerUrl = container.Uri.AbsoluteUri + containerSas;
+            job.CommonEnvironmentSettings = new[] { new EnvironmentSetting("JOB_CONTAINER_URL", containerUrl) };
+
+            // Commit the job to the Batch service
+            job.Commit();
+            Console.WriteLine($"Created job {jobId}");
+
+            // Obtain the bound job from the Batch service
+            job = batchClient.JobOperations.GetJob(jobId);
+
+            IEnumerable<CloudTask> tasks = Enumerable.Range(1, 20).Select(i =>
+                new CloudTask(i.ToString().PadLeft(3, '0'), $"cmd /c %AZ_BATCH_APP_PACKAGE_{appPackageId.ToUpper()}#{appPackageVersion}%\\PersistOutputsTask.exe")
+            );
+
+            // Add the tasks to the job; the tasks are automatically
+            // scheduled for execution on the nodes by the Batch service.
+            job.AddTask(tasks);
+
+            Console.WriteLine($"All tasks added to job {job.Id}");
+            Console.WriteLine();
+
+            foreach (CloudTask task in Program.CompletedTasks(job))
```

Comment:
If you're refactoring like this then move `CompletedTasks()` out of the Program class and put it into a meaningfully named class or make it an extension method on CloudJob.  (It might need to be made async too.)